model/src/main/java/com/microsoft/java/bs/gradle/model/JavaExtension.java

```diff
@@ -11,10 +11,6 @@
  * The extension model for Java language.
  */
 public interface JavaExtension extends Serializable {
```

Comment:
Do you think if it makes sense that creating an interface called something like `LanguageExtension` and has a method called `convert()`.

Each language extension extends this interface and implemented the `convert` by their own?

Reply:
@jdneo I've added another commit to push all conversions (and reflection use) into the LanguageExtension.